libs/designsystem/src/lib/components/calendar/calendar.component.ts

```diff
@@ -196,46 +253,41 @@ export class CalendarComponent implements OnInit, AfterViewInit, OnChanges {
   // We currently log no warnings if the date doesn't match the timezone setting or
   // if it doesn't point to midnight.
   private normalizeDate(dateLocalOrUTC: Date) {
-    if (dateLocalOrUTC) {
-      // prettier-ignore
-      if (moment(dateLocalOrUTC).startOf('day').isSame(moment(dateLocalOrUTC))) {
-        // date is local timezone midnight
-        return dateLocalOrUTC;
-      
-      // prettier-ignore
-      } else if (moment.utc(dateLocalOrUTC).startOf('day').isSame(moment.utc(dateLocalOrUTC))) {
-        // the date is a utc midnight; create the equivalent local timezone midnight date
-        const utcMidnightMoment = moment.utc(dateLocalOrUTC);
-        return moment(utcMidnightMoment.format('YYYY-MM-DD')).toDate();
-      
-      } else {
-        // does not point to midnight so best assumption is to chop off the time part
-        return moment(dateLocalOrUTC).startOf('day').toDate();
-      }
-    }
-  }
+    if (!dateLocalOrUTC) return;
 
-  private normalizeDates(datesLocalOrUTC: Date[]) {
-    if (datesLocalOrUTC) {
-      return datesLocalOrUTC.map(this.normalizeDate);
+    if (startOfDay(dateLocalOrUTC).getTime() === dateLocalOrUTC.getTime()) {
+      // date is local timezone midnight
+      return dateLocalOrUTC;
+    }
+    if (
+      startOfDay(utcToZonedTime(dateLocalOrUTC, this.timeZoneName)).getTime() ===
+      utcToZonedTime(dateLocalOrUTC, this.timeZoneName).getTime()
+    ) {
+      // the date is a UTC midnight; create the equivalent local timezone midnight date
+      const normalizedUTCdate = utcToZonedTime(dateLocalOrUTC, this.timeZoneName);
+      return normalizedUTCdate;
     }
+    // does not point to midnight so we make it
+    return startOfDay(dateLocalOrUTC);
   }
 
-  private capitalizeFirstLetter(string: string) {
-    if (typeof string !== 'string') {
-      return '';
-    }
-    return string.charAt(0).toUpperCase() + string.slice(1);
+  private normalizeDates(datesLocalOrUTC?: Date[]) {
+    return (datesLocalOrUTC || []).map((date) => this.normalizeDate(date));
   }
 
   private getWeekDays(): string[] {
-    return [1, 2, 3, 4, 5, 6, 7].map((index) =>
-      moment()
-        .isoWeekday(index)
-        .format('dd')
-        .substr(0, 1)
-        .toUpperCase()
-    );
+    const now = new Date();
+
+    const weekInterval = eachDayOfInterval({
+      start: startOfWeek(now, { locale: this.locale }),
+      end: endOfWeek(now, { locale: this.locale }),
+    });
```

Comment:
This now takes into account the first day of week based in locale, where as the old method used isoWeekDay - this might align wrong compared with the dates array used to generate the days table...

Reply:
It did 😄 
Now fixed, so we don't use ISO week anywhere